### DISCUSSION TITLE
Doesn't work for workspace / common dependencies.

### DISCUSSION THREAD

**Discussion by foges** [webpack/webpack#17977] [Q&A]
# Bug report

As far as I can tell webpack currently doesn't work very well in monorepos where projects depend on each other.

**What is the current behavior?**

Assume you have 2 packages in your monorepo

 - `monorepo/packages/common`
 - `monorepo/packages/project`        <- Webpack project that has `common` as a dependency

The monorepo/package.json has
```
"workspaces": [
  "packages/common",
  "packages/project",
],
```

and `monorepo/packages/project/package.json` has

```
"dependencies": {
  "@monorepo/common": "1.0.0",
}
```

As part of the build process webpack will create a package.json in `.webpack/dependencies` with the minimal dependencies it needs and runs `yarn install --non-interactive --frozen-lockfile`. Now because `monorepo/packages/common/.webpack/dependencies` isn't in the list of workspaces in `monorepo/package.json` yarn will not look for  `@monorepo/common` in the workspace, and instead tries to find it on yarnpkg.org, where it doesn't exist, resulting in `
error An unexpected error occurred: "https://registry.yarnpkg.com/@monorepo/common: Not found".`.

You can get around it by adding the webpack directory, but that seems very hacky / brittle

```
"workspaces": [
  "packages/common",
  "packages/project",
  "packages/common/.webpack/dependencies"        <- new
],
```

**If the current behavior is a bug, please provide the steps to reproduce.**

See steps above

**What is the expected behavior?**

Webpack bundling should work in a monorepo with shared dependencies

**Other relevant information:**
webpack version:  webpack@5.88.2
Node.js version: v18.16.0
Yarn version: v1.22.19
Operating System: Mac OS Ventura 13.4.1
Additional tools: 

-----

NB: I'm doing this through the Serverless Framework, so there's a chance it's specific to that, although from what I can tell it's a webpack issue.

**Comment by alexander-akait** [comment]
Sorry for delay, can you provide reproducible example?
